server/src/main/java/io/crate/execution/dml/ObjectIndexer.java

```diff
@@ -94,41 +92,29 @@ public ObjectIndexer(RelationName table,
     }
 
     @Override
-    public void indexValue(@NotNull Map<String, Object> value,
-                           Consumer<? super IndexableField> addField,
-                           TranslogWriter translogWriter,
-                           Synthetics synthetics,
-                           Map<ColumnIdent, Indexer.ColumnConstraint> checks) throws IOException {
+    public void indexValue(@NotNull Map<String, Object> value, IndexDocumentBuilder docBuilder) throws IOException {
+        TranslogWriter translogWriter = docBuilder.translogWriter();
         translogWriter.startObject();
         for (var entry : children.entrySet()) {
             String innerName = entry.getKey();
             Child child = entry.getValue();
             Object innerValue = null;
-            if (value == null || value.containsKey(innerName) == false) {
-                Input<Object> synthetic = synthetics.get(child.ident());
-                if (synthetic != null) {
-                    innerValue = synthetic.value();
-                }
+            if (value.containsKey(innerName) == false) {
```

Comment:
`value == null` is gone, just double checking.

Reply:
Yes, the contract on ValueIndexer.indexValue() is that `null` is never passed as a value.  Nulls are handled in Indexer, ArrayIndexer or ObjectIndexer before they are passed down to the relevant ValueIndexer.